internal/deepprint/deepprint.go

```diff
@@ -12,16 +12,27 @@
 package deepprint
 
 import (
+	"bytes"
 	"crypto/sha256"
 	"fmt"
 	"io"
 	"reflect"
 )
 
-func Hash(v interface{}) string {
+func Hash(v interface{}) []byte {
 	h := sha256.New()
 	Print(h, v)
-	return fmt.Sprintf("%x", h.Sum(nil))
+	return h.Sum(nil)
+}
+
+// UpdateHash sets last to the hash of v and returns whether its value changed.
```

Comment:
The usual Go wording for boolean funcs is "reports whether" instead of "returns whether" (minor)